### PR TITLE
Create Signer Container

### PR DESCRIPTION
**Pull request by alextserepov**
- Docker container with tools
  - USB rules for accessing Yubikey
  - openssl config for Yubikey

### CODE REVIEW

File: yubikey/start.sh [hunk 1]
```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
```
Comment (cazfi) [inline yubikey/start.sh:1]: I would either move files that get copied inside the container in a subdirectory, or document the situation by providing a .dockerignore -file that lists these files (this file) that are to be used on host side (only).

File: README.md [hunk 2]
```diff
@@ -1,2 +1,36 @@
 # scs-pki-research
```
Comment (jenninikko) [inline README.md:1]: An empty line is needed after the heading and a period at the end of the sentence.
Please check other headings, it is a good rule to wrap a heading with lines: two newlines before and one after.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,36 @@
...
 PKI research material
+
+# Example signing:
```
Comment (jenninikko) [inline README.md:4]: Let's get rid of a colon here.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,36 @@
...
+
+Generate SHA256 hash
```
Comment (jenninikko) [inline README.md:6]: Please put a colon here.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,36 @@
...
+
+`$ sha256sum 0r434a8n04i14s4sp3fam23cplwyn1iw-generic-x86_64-debug-nixos.img 
+1e648d85a8ca1c6f55fcc060bbcbd24a4ff2deb5844e198088689ca840f3337a  0r434a8n04i14s4sp3fam23cplwyn1iw-generic-x86_64-debug-nixos.img`
+
+Sign the hash with Yubikey
```
Comment (jenninikko) [inline README.md:11]: Sign the hash with the Yubikey:

File: README.md [hunk 6]
```diff
@@ -1,2 +1,36 @@
...
+
+`$ ./start.sh --sign --h=1e648d85a8ca1c6f55fcc060bbcbd24a4ff2deb5844e198088689ca840f3337a
+YubiKey found on USB Bus: 003, Device: 002
+499OENQ5B08XVLL4nBJdeoWfYl2TbQEatp1SxeDMnqf527ie6OVBrNCh9SmfCD24rw6v/C/QQ13OOBHBehKQEA==`
+
+Verify the signature (Valid case)
```
Comment (jenninikko) [inline README.md:17]: 1. Verify the signature (valid case):
2. Add an empty line before and after the code line.

File: README.md [hunk 7]
```diff
@@ -1,2 +1,36 @@
...
+`alex@alex-unikie:~/repos/scs-pki-research/yubikey$ ./start.sh --verify --h=1e648d85a8ca1c6f55fcc060bbcbd24a4ff2deb5844e198088689ca840f3337a --sg=499OENQ5B08XVLL4nBJdeoWfYl2TbQEatp1SxeDMnqf527ie6OVBrNCh9SmfCD24rw6v/C/QQ13OOBHBehKQEA==
```
Comment (jenninikko) [inline README.md:18]: Needs ` at the end.

File: README.md [hunk 8]
```diff
@@ -1,2 +1,36 @@
...
+YubiKey found on USB Bus: 003, Device: 002
```
Comment (jenninikko) [inline README.md:19]: The YubiKey is found on a USB, Bus: 003, Device: 002.

File: README.md (hunk 2 above)
Comment (jenninikko) [inline README.md:1]: SPDX license identifier?

**Review by jenninikko — COMMENTED** [review]
I will be a little tedious)
A few comments from my side.

**Review by tervis-unikie — CHANGES_REQUESTED** [review]
Please squash your commits. 
This is difficult to review having so many changes in many commits. 
And we don't want to merge a bunch commits doing things slightly wrong and then bunch of commits fixing the things, when we can have only commits that just do things right. :) 

Just reminding that, it is okay to force push into PR branches when doing changes.

Maybe while you're at it you might want to combine some docker RUN steps as much as possible, by concatenating commands with "&&". This will require less intermediate containers.

File: yubikey/start.sh [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+    echo ""
+    echo "  Options:"
+    echo "    --sign      = Sign the binary using the hash"
+    echo "    --verify    = Verify the signature"
+    echo "    --h=HSH     = Set the hash for signing or verification"
+    echo "    --sg=SGN    = Set the signature for verification (b64 format)"
+    echo ""
+    echo "  Example:"
+    echo ""
+    echo "$0 --sign -h 35b5d3b9746967f4c1984f90d9ccccda6862abd8dc2546a9b8c663c610696617 > signature.b64"
```
Comment (cazfi) [inline yubikey/start.sh:20]: Examples have wrong syntax. Should be "--h=???" (not "-h ???" or "--h ???")

**Pull request merged**